WebApi/WebApi/Controllers/PostController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using WebApi.Models.DTO;
+using WebApi.Models.DTO.PostDTOs;
+using WebApi.Services.Services_Interfaces;
+
+
+namespace WebApi.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class PostController : ControllerBase
+    {
+        private readonly IPostService _postService;
+        private readonly ILogger<PostController> _logger;
+        public PostController(ILogger<PostController> logger , IPostService postService)
+        {
+            _logger = logger;
+            _postService = postService;
+        }
+
+
+        [HttpGet]
+        public ActionResult<IQueryable<PostDTO>> GetAll([Required][FromHeader] int userID)
+        {
+            var result = _postService.GetAll();
+            if (result != null) return Ok(result);
+            else return NotFound();
+        }
+
+      
+        //[HttpGet("{UserID}")]
+        [HttpGet]
+        public ActionResult<IQueryable<PostDTO>> GetUserPosts([Required][FromQuery] int UserID) //[FromHeader] int headerID
+        {
+            var result = _postService.GetAllOfUser(UserID);
+            if (result != null) return Ok(result);
+            else return NotFound();
+        }
+
+
+        [HttpGet("{postID}")]
+        public ActionResult<PostDTO> Get([Required][FromHeader] int userID, [FromRoute] int postID)
+        {
+            var result = _postService.GetById(postID);
+            if (result != null) return Ok(result);
+            else return NotFound();
+        }
+
+
+        [HttpDelete("{postID}")]
+        public ActionResult<Task> Delete([Required][FromHeader] int userID, [FromRoute] int postID)//, [FromBody] DateTime dateTime) - nie wiem czemu z tym nie działa. Może trzeba zoribć DTO na datetime?
+        {
+            _postService.DeletePostAsync(postID);
+            return Ok();
```

Comment:
nie wydaje ci się dziwne, że zawsze zwracamy Ok?
Przecież możemy chcieć usunąć post o id, którego nie ma, no nie?
Zobacz co się wtedy dzieje:
serwis najpierw szuka id w bazie (getByID) i zwraca nulla i przekazuje go do metody DeleteAsync,
która przy nullu rzuca wyjątek, którego nikt nie łapie, tylko go ignorujesz i zwracasz Ok() XD

W serwisie, jeżeli RemoveAsync może rzucić wyjątek to powinno być w bloku try.. catch
w catch (ale dla ArgumentNullException -> czyli wyjątka rzucanego przy braku danego posta w db, bo ta metoda rzuca też inny wyjątek !!!) zwróć nulla, a w kontrolerze dodaj że jeżeli null to NotFound()